### DISCUSSION TITLE
Queuing/setting a cookie and retrieving it in the same request

### DISCUSSION THREAD

**Discussion by booni3** [livewire/livewire#1787] [1. Help] [answered]
I am using cookies to set some computer-specific settings, such as which printer that workstation is using. For V1 I had simply:

```php
public function setPrinter($slug)
{
    Cookie::queue('printer_name', $slug);
}
```

This worked find. You click the `<button wire:click="setPrinter('printer1')">Set Printer</button>` button and the cookie would be set and at the same time you would see the visible change on the page:

```blade
@if(Cookie::get('printer_name')=='printer 1') 
printer 1 selected 
@endif
```

Since V2, the cookie queue function no longer sets the cookie prior to rendering the page. So while it is still technically set, the blade view is not updated to show so.

I have tried to do something like refresh or re-render after the cookie is queued but this does not work either. Does anyone have any ideas?

**Comment by austenc** [comment, marked as answer]
@booni3 This is because `Cookie::queue` sets the cookie on the first response, but as you noted, you cannot retrieve it until making a second request. However, I think we can easily work around this!

You should store the printer state in the component and use a [Lifecycle Hook](https://laravel-livewire.com/docs/2.x/lifecycle-hooks) to update the cookie. This way the cookie is set, and you're referencing state from the Livewire component itself, which I believe is a better practice anyway. 

Try this out, I think it should solve it for you:

**Component**
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;
use Illuminate\Support\Facades\Cookie;

class Example extends Component
{
    public $printer; 

    public function mount()
    {
        $this->printer = Cookie::get('printer_name');
    }

    public function updatedPrinter($slug)
    {
        Cookie::queue('printer_name', $slug);
    }

    public function render()
    {
        return view('livewire.example');
    }
}
```

**View**
```
<div>
    <button wire:click="$set('printer', 'Some Printer')">Set Printer</button>
    Printer is: {{ $printer }}
</div>
```

**Reply by booni3** [reply]
This works well. Thanks for the suggestion

**Reply by mostafa8722** [reply]
Thanks a lot. It worked for me.

**Comment by calebporzio** [comment]
Thanks @austenc for digging into this.

If more people run into this behavior change from v1 to v2 - please submit an issue about it.
